src/pipeline_v3/utils/security.py

```
@@ -0,0 +1,462 @@
+
+        Raises:
+            SecurityError: If path validation fails
+        """
+        # Convert to Path and resolve to absolute path
+        path_obj = Path(path).expanduser().resolve()
+
+        # Default to current working directory if no base dirs specified
+        if allowed_base_dirs is None:
+            allowed_base_dirs = [Path.cwd()]
+            # Also allow temp directories for testing
+            import tempfile
```

Pylint (3.3.7)

[convention] Import outside toplevel (tempfile)

(C0415)